providers/bind/soa.go

```diff
@@ -19,10 +23,22 @@ func makeSoa(origin string, defSoa *SoaDefaults, existing, desired *models.Recor
 		desired = &models.RecordConfig{}
 	}
 
+	soaMail := firstNonNull(desired.SoaMbox, existing.SoaMbox, defSoa.Mbox, "DEFAULT_NOT_SET.")
```

Comment:
This should be a separate function, with unit tests.

Reply:
Is there a good place to globally declare this function, so all providers can use it? (Additionally, which function name would you think would be good here?)

Reply:
I can't find an existing place that is good. However there is `pkg/txtutil` so why not create `pkg/soautil`?

Suggestion: create `pkg/soautil/soautil.go` and put the function in there.

Reply:
Done